build/pmm-client-docker/entrypoint.py

```diff
@@ -77,13 +79,19 @@ def main():
 
         agent.wait()
 
-        if status != 0:
+        if status != 0 and not PMM_AGENT_SIDECAR:
             sys.exit(status)
 
-    # use execlp to replace the current process (this entrypoint)
-    # with pmm-agent with inherited environment
-    print('Starting pmm-agent ...', file=sys.stderr)
-    os.execlp('pmm-agent', 'run')
+    # restart pmm-agent if PMM_AGENT_SIDECAR flag is provided
+    while True:
+        print('Starting pmm-agent ...', file=sys.stderr)
+        status = subprocess.call(['pmm-agent', 'run'])
```

Comment:
To be honest I can see a potencial problem here. After this change, PID 1 in the container will be entrypoint.py but we don't have any signal handler here and Kubernetes will be forced to kill the container with the 9th signal (SIGKILL).
Here are details: https://petermalmgren.com/signal-handling-docker/

Also, there are no cases when we want to restart the container?
Restart is also an auto-healing feature and sometimes it can help to fix issues (for example if one of the Kubernetes node is lost connectivity). Maybe we can restart the container after 100 fails for example?

Reply:
Good point, will fix it.

Reply:
Fixed, PTAL